api/src/main/java/org/openmrs/api/EncounterService.java

```diff
@@ -724,4 +725,24 @@ public Integer getEncountersByVisitsAndPatientCount(Patient patient, boolean inc
 	 */
 	@Authorized( { PrivilegeConstants.EDIT_ENCOUNTERS })
 	public Encounter transferEncounter(Encounter encounter, Patient patient);
+
+	/**
+	 * Get EncounterRoles by name
+	 *
+	 * @param name
+	 * @return List of EncounterRole objects
+	 * @since 1.11
+	 * @should find encounter roles based on their name
+	 */
+
+	/**
+	 * Check if encounter has a particular diagnosis
+	 * 
+	 * @param encounter
+	 * @param diagnosis
+	 * 
+	 * @return check if the encounter has a particular diagnosis
+	 */
+	@Authorized( { PrivilegeConstants.GET_ENCOUNTER_ROLES })
```

Comment:
I thought you were going to put this in the Encounter class

Reply:
I do not know why I should put it in the Encounter class. More clarifications on that please.
I added it here because all other methods of the EncounterService class have this annotation here.

Reply:
If the call resulted into accessing the database, then that would go via the service later. But your implementation seems to just loop through the diagnosis list that the encounter object already has.

Reply:
Noted. Reverted